src/core/server/http/security_response_headers_config.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+const strictTransportSecuritySchema = schema.object({
+  // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+  maxAge: schema.duration({ defaultValue: '1Y' }),
+  includeSubDomains: schema.boolean({ defaultValue: false }),
+  preload: schema.boolean({ defaultValue: false }),
+});
```

Comment:
By default, `Strict-Transport-Security` is enabled with a max age of 1 year. This is recommended by MDN and most other authoritative sources. We _could_ decrease this but I'm not sure we should.

I omitted `includeSubDomains` by default because of https://github.com/elastic/kibana/issues/52809#issuecomment-564764345.

Note: if Kibana is served over HTTP, this header is still sent, but browsers ignore it. It only affects browsers once they see this header in an HTTPS response.